Simplenote/Controllers/AccountDeletionController.swift

```diff
@@ -2,18 +2,24 @@ import Foundation
 
 @objc
 class AccountDeletionController: NSObject {
-    private let accountDeletionRequestDate = Date()
+    private var accountDeletionRequestDate: Date?
 
     var deletionTokenHasExpired: Bool {
-        guard let expirationDate = accountDeletionRequestDate.increased(byDays: 1) else {
+        guard let requestDate = accountDeletionRequestDate,
+              let expirationDate = requestDate.increased(byDays: 1) else {
             return true
         }
 
         return Date() > expirationDate
     }
 
     func requestAccountDeletion(_ user: SPUser, completion: @escaping (_ result: Result<Data?, RemoteError>) -> Void) {
-
-        AccountRemote().requestDelete(user, completion: completion)
+        accountDeletionRequestDate = Date()
+        AccountRemote().requestDelete(user) { [weak self] (result) in
+            if case .success = result {
+                self?.accountDeletionRequestDate = Date()
```

Comment:
Curious why do we need to set `accountDeletionRequestDate` twice? 
I think it should be enough to do it only when we get a successful result?

Reply:
Ha!  We don't.  I had it running when the user requested the delete at first, but changed it to if a success is received and must have forgotten to remove the first one